alacritty/src/input.rs

```diff
@@ -1002,7 +1002,33 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
     /// Process key input.
     pub fn key_input(&mut self, key: KeyEvent) {
         // IME input will be applied on commit and shouldn't trigger key bindings.
-        if key.state == ElementState::Released || self.ctx.display().ime.preedit().is_some() {
+        if self.ctx.display().ime.preedit().is_some() {
+            return;
+        }
+
+        let mode = *self.ctx.terminal().mode();
+        let mods = self.ctx.modifiers().state();
+
+        if key.state == ElementState::Released {
+            if mode.contains(TermMode::KEYBOARD_REPORT_EVENT_TYPES)
+                && !mode.contains(TermMode::VI)
+                && !self.ctx.search_active()
+                && !self.ctx.display().hint_state.active()
+            {
+                // NOTE: echoing the key back on release is how it's done in kitty/foot and
+                // it's how it should be done according to the kitty author
+                // https://github.com/kovidgoyal/kitty/issues/6516#issuecomment-1659454350
+                let bytes: Cow<'static, [u8]> = match key.logical_key.as_ref() {
+                    Key::Tab => [b'\t'].as_slice().into(),
+                    Key::Enter => [b'\r'].as_slice().into(),
+                    Key::Delete => [b'\x7f'].as_slice().into(),
+                    Key::Escape => [b'\x1b'].as_slice().into(),
```

Comment:
I need to do it manually for `Backspace`(because winit has different code) and I don't want to deal with `Option` from invoking `to_text`, even though I can unwrap it.